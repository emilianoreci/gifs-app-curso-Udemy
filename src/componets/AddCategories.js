//  @Package
import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategories = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("handleSubmit--llamado, ", inputValue); para el test

    //En vez de recibir las categories por props
    //setCategories([...categories, inputValue]);
    if (inputValue.trim().length > 2) {
      setCategories((cate) => [inputValue, ...cate]); //es un set de useState()
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <h2>AddCategories</h2>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategories.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
