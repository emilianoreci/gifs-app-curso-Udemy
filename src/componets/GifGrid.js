//  @Package
import React from "react";
import { PropTypes } from "prop-types";

//  @Own
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  //hago destructuring y le cambio el nombre de data a images para usarlo.
  const { data: images, loading } = useFetchGifs(category);
  return (
    <>
      <h2 className="animate__animated animate__fadeIn">{category}</h2>
      <div className="card-grid">
        {loading && <p>cargando...</p>}
        {images.map((eachElement) => (
          <GifGridItem key={eachElement.id} {...eachElement}></GifGridItem>
          /* <GifGridItem key={eachElement.id} img={eachElement}></GifGridItem> */
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
