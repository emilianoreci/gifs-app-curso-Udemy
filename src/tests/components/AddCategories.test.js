import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { AddCategories } from "../../componets/AddCategories";

describe("Pruebas al componente AddCategories", () => {
  //const setCategories = () => {}; // manera provisoria para testear una fn.
  const setCategories = jest.fn(); //es una fn
  let wrapper = shallow(<AddCategories setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategories setCategories={setCategories} />);
  });

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar el valor del input de texto", () => {
    const input = wrapper.find("input");
    const value = "hola mundo";

    //el segundo argumento de simulate, se pone {} que seria el e(event)
    //input.simulate('change', {})
    //al e.target.value
    input.simulate("change", { target: { value: value } });
    expect(wrapper.find("p").text().trim()).toEqual(value);
  });

  test("No debe de postear la informacion con el submit", () => {
    //provee los argumentos que se transformaran en el evento, en Ã©ste caso el e.preventDefault
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar a setCategories() y limpiar la caja de texto", () => {
    const value = "banana";

    // simular el inputChange
    wrapper.find("input").simulate("change", { target: { value: value } });

    // simular el submit
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    // se debe de haber llamado setCategories
    expect(setCategories).toHaveBeenCalled();
    //.toHaveBeenCalledTimes(3) se espera que llame a setCategories 3 veces
    //expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); //espera q se llame a la fn setCategories y que se le pase como argumento una funcion.

    // el valor de input debe ser ''
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
