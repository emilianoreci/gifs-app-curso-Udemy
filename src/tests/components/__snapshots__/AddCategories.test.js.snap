// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente AddCategories debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    AddCategories
  </h2>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
